go/vt/topo/cell_info.go

```diff
@@ -176,37 +176,52 @@ func (ts *Server) GetKnownCells(ctx context.Context) ([]string, error) {
 // ExpandCells takes a comma-separated list of cells and returns an array of cell names
 // Aliases are expanded and an empty string returns all cells
 func (ts *Server) ExpandCells(ctx context.Context, cells string) ([]string, error) {
-	var err error
-	var outputCells []string
-	inputCells := strings.Split(cells, ",")
+	var (
+		err         error
+		inputCells  []string
+		outputCells = sets.NewString() // Use a set to dedupe if the input cells list includes an alias and a cell in that alias.
+	)
+
 	if cells == "" {
 		inputCells, err = ts.GetCellInfoNames(ctx)
 		if err != nil {
 			return nil, err
 		}
+	} else {
+		inputCells = strings.Split(cells, ",")
 	}
 
-	for _, cell := range inputCells {
-		cell2 := strings.TrimSpace(cell)
+	expandCell := func(ctx context.Context, cell string) error {
 		shortCtx, cancel := context.WithTimeout(ctx, *RemoteOperationTimeout)
 		defer cancel()
-		_, err := ts.GetCellInfo(shortCtx, cell2, false)
+
+		_, err := ts.GetCellInfo(shortCtx, cell, false /* strongRead */)
 		if err != nil {
-			// not a valid cell, check whether it is a cell alias
+			// Not a valid cell name. Check whether it is an alias.
 			shortCtx, cancel := context.WithTimeout(ctx, *RemoteOperationTimeout)
 			defer cancel()
-			alias, err2 := ts.GetCellsAlias(shortCtx, cell2, false)
-			// if we get an error, either cellAlias doesn't exist or it isn't a cell alias at all. Ignore and continue
-			if err2 == nil {
-				outputCells = append(outputCells, alias.Cells...)
-			}
-			if err != nil {
```

Comment:
For my understanding, here is the bug in the current code. Per conversation with @ajm188 
> we don't nil out err if err2 is nil, so even if there's a valid alias, we still return the error from trying to look up the alias as pure cell